pmd-core/src/main/java/net/sourceforge/pmd/SourceCodeProcessor.java

```diff
@@ -113,7 +113,9 @@ private void processSourceCodeWithoutCache(final Reader sourceCode, final RuleSe
             processSource(sourceCode, ruleSets, ctx);
         } catch (ParseException pe) {
             configuration.getAnalysisCache().analysisFailed(ctx.getSourceCodeFile());
-            throw new PMDException("Error while parsing " + ctx.getSourceCodeFile(), pe);
+            if (!configuration.isForceLanguageVersion()) {
```

Comment:
That means, if force-language is set, and we fail to parse a file, we silently ignore it.... that makes the noise quiet if there are many other files and we analyze all files in the directory. But it will also ignore the parsing error for a file, which should be valid....

I don't think, we can easily solve this (otherwise we wouldn't need force-language at all if we knew which files to scan). But it's worth to document that feature that you never see a parse error once you use "force-language". The only effect will be, that you might loose violations.

Reply:
I guess, at least logging will be necessary

Reply:
Agreed - That was my thought process, too. I guess this would be something to be documented in the user docs rather than the code? 

I tend to think of PMD as something I only run on code that is syntactically correct according to my compiler/interpreter, so I don't think you lose much by being silent on parse errors.